src/main/java/com/mongodb/kafka/connect/sink/MongoSinkTopicConfig.java

```diff
@@ -551,7 +551,10 @@ public Optional<WriteModelStrategy> getDeleteWriteModelStrategy() {
                   PartialKeyStrategy.class.getSimpleName(),
                   ProvidedInKeyStrategy.class.getSimpleName()));
         }
-        deleteOneWriteModelStrategy = new DeleteOneDefaultStrategy(idStrategy);
+      } else if (!getString(DELETE_WRITEMODEL_STRATEGY_CONFIG).isEmpty()) {
```

Comment:
This can just be an else as the validator will ensure its not an empty string.

Reply:
you're right
If it doesn't fit FULLY_QUALIFIED_CLASS_NAME it will fail.
However, if the user does not set delete.write.model, the default value is applied and the appropriate genetaredId method according to the id strategy is used.